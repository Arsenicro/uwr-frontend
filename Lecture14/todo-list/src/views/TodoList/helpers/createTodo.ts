import { v4 } from 'uuid';

import { ITodo } from '../../../types/Todo.types';

interface IProps {
  name: string;
}

export const createTodo = ({name}: IProps): ITodo => ({
  id: v4(),
  name,
  completed: false
});
