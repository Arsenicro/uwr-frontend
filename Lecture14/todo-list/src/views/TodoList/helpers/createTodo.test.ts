import { describe, expect, it } from 'vitest';

import { createTodo } from './createTodo';

describe("Create Todo", () => {
  /* it("Should create Todo with Name", () => {
    // Arrange
    const name = "New Todo";

    // Act
    const todo = createTodo({name: name});

    // Asset
    expect(todo.name).toEqual(name);
  });

  it("Should return unique id", () => {
    // Arrange
    const name1 = "New Todo";
    const name2 = "New Todo";

    // Act
    const todo1 = createTodo({name: name1});
    const todo2 = createTodo({name: name2});

    // Asset
    expect(todo1.id).not.toEqual(todo2.id);
  }); */

  it("Should create Todo with Name", () => {
    // Arrange
    const name = "New Todo";

    // Act
    const todo = createTodo({name: name});

    // Asset
    expect(todo.name).toMatchSnapshot();
  });
});
