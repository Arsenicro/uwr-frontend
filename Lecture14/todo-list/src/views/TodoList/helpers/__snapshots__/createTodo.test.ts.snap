// Vitest Snapshot v1

exports[`Create Todo > Should create Todo with Name 1`] = `"New Todo"`;
